File: PincruxOBJSample/PincruxOBJSample/PubkeyViewController.m
//
//  PubkeyViewController.m
//  PincruxOBJSample
//
//  Created by K_JINY on 2022/07/25.
//

#import <Foundation/Foundation.h>
#import "PubkeyViewController.h"

@interface PubkeyViewController ()

@end

@implementation CustomNavi
- (UIInterfaceOrientationMask)supportedInterfaceOrientations {
    return UIInterfaceOrientationMaskPortrait;
}
@end

@implementation PubkeyViewController


- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.pubkeyF.text = [self getCustomPubkey];
}

- (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event {
    [self.view endEditing:YES];
}

- (void)setCustomPubkey:(NSString *)pubkey {
    [[NSUserDefaults standardUserDefaults] setObject:self.pubkeyF.text forKey:@"offerwallPubKey"];
    [[NSUserDefaults standardUserDefaults] synchronize];
}

- (NSString *)getCustomPubkey {
    NSString *customPubkey = [[NSUserDefaults standardUserDefaults] stringForKey:@"offerwallPubKey"];
    return customPubkey;
}

- (IBAction)customStartBtt:(id)sender {
    if([self.pubkeyF.text length] == 0) {
        UIAlertController * alert = [UIAlertController alertControllerWithTitle:@"PubKey를 입력하세요"
                                                                        message:nil preferredStyle:UIAlertControllerStyleAlert];
        
        UIAlertAction* confirm = [UIAlertAction actionWithTitle:@"확인" style:UIAlertActionStyleDefault handler:^(UIAlertAction * action) {
            [alert dismissViewControllerAnimated:YES completion:nil];
        }];
        
        [alert addAction:confirm];
        [self presentViewController:alert animated:YES completion:nil];
    }else {
        [self setCustomPubkey:self.pubkeyF.text];
        
        SetOfferwallViewController *settingVC = [self.storyboard instantiateViewControllerWithIdentifier:@"setOfferwallVC"];
        settingVC.pubkey = self.pubkeyF.text;
        [self.navigationController pushViewController: settingVC animated:YES];
    }
}
@end

File: PincruxSwiftSample/PincruxSwiftSample/PubkeyViewController.swift
//
//  PubkeyViewController.swift
//  PincruxSwiftSample
//
//  Created by K_JINY on 2022/07/22.
//

import UIKit

class PubkeyViewController: UIViewController {
    @IBOutlet weak var pubkeyfeild: UITextField!

    override func viewDidLoad() {
        super.viewDidLoad()
        self.pubkeyfeild.text = getCustomKey()
        // Do any additional setup after loading the view.
    }
    
    @IBAction func nextBtt(_ sender: Any) {
        self.view.endEditing(true)
        setCustomKey(pubkeyfeild.text ?? "")
        let setOfferwallVC = self.storyboard?.instantiateViewController(withIdentifier: "setOfferwallVC") as? SettingViewController ?? SettingViewController()
        setOfferwallVC.pubkey = pubkeyfeild.text ?? ""
        self.navigationController?.pushViewController(setOfferwallVC, animated: true)
    }
    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        self.view.endEditing(true)
    }
    
    func setCustomKey(_ key: String) {
        if key.count == 0 { return }
        UserDefaults.standard.setValue(key, forKey: "offerwallPubKey")
    }
    
    func getCustomKey() -> String {
        guard let customKey = UserDefaults.standard.string(forKey: "offerwallPubKey") else {
            return ""
        }
        return customKey
    }
}
